webpack/scenes/Subscriptions/components/SubscriptionsTable/__tests__/SubscriptionsTableHelpers.test.js

```diff
@@ -0,0 +1,123 @@
+import groupedSubscriptions from './SubscriptionsTable.fixtures';
+import { buildTableRows } from '../SubscriptionsTableHelpers';
+
+describe('Build Table Rows', () => {
+  it('should display correct maxQuantity', () => {
+    const availableQuantities = {
+      1: 50,
+      2: 50,
+      3: -1,
+      4: 100,
+      5: 50,
+    };
+
+    const rows = [
+      {
+        consumed: 0,
+        id: 1,
+        maxQuantity: 60,
+        name: 'Alpha',
+        product_id: 'RH00001',
+        quantity: 10,
+        upstreamAvailable: 50,
+        upstreamAvailableLoaded: true,
+      },
+      {
+        consumed: 0,
+        id: 3,
+        maxQuantity: -1,
+        name: 'Charlie',
+        product_id: 'RH00002',
+        quantity: 10,
+        upstreamAvailable: -1,
+        upstreamAvailableLoaded: true,
+      },
+      {
+        consumed: 0,
+        id: 4,
+        maxQuantity: 200,
+        name: 'Bravo',
+        product_id: 'RH00003',
+        quantity: 100,
+        upstreamAvailable: 100,
+        upstreamAvailableLoaded: true,
+      },
+      {
+        consumed: 5,
+        id: 5,
+        maxQuantity: 65,
+        name: 'Delta',
+        product_id: 'RH00004',
+        quantity: 15,
+        upstreamAvailable: 50,
+        upstreamAvailableLoaded: true,
+      },
+    ];
+
+    expect(buildTableRows(groupedSubscriptions, availableQuantities, {}))
+      .toEqual(rows);
+  });
+
+  it('should update quantities', () => {
+    const availableQuantities = {
+      1: 50,
+      2: 50,
+      3: -1,
+      4: 100,
+      5: 50,
+    };
+
+    const updatedQuantities = {
+      1: 20,
+      4: 150,
+    };
+
+    const rows = [
```

Comment:
Seems like there is a small difference between this and the other rows object array. You could DRY this up doing something like the following. I would also put these in the fixtures.

```javascript
const subOne =       {
        consumed: 0,
        id: 1,
        maxQuantity: 60,
        name: 'Alpha',
        product_id: 'RH00001',
        quantity: 10,
        upstreamAvailable: 50,
        upstreamAvailableLoaded: true,
      };

const subTwo =       {
        consumed: 0,
        id: 3,
        maxQuantity: -1,
        name: 'Charlie',
        product_id: 'RH00002',
        quantity: 10,
        upstreamAvailable: -1,
        upstreamAvailableLoaded: true,
      };
// ... same for the other two

// inside first test
const rows = [subOne, subTwo, subThree, subFour];

// inside second test
const rows = [ {...subOne, entitlementsChanged: true, quantity: 20 }, subTwo, { ...subThree, entitlementsChanged: true, quantity: 150 }, subFour];
```